ako-gateway-api/lib/control_config.go

```diff
@@ -0,0 +1,178 @@
+		akoControlConfigInstance = &akoControlConfig{
+			controllerVersion: initControllerVersion(),
+		}
+	}
+	return akoControlConfigInstance
+}
+
+func (c *akoControlConfig) SetGatewayAPIClientset(cs gwApi.Interface) {
+	c.gwApiCS = cs
+}
+
+func (c *akoControlConfig) GatewayAPIClientset() gwApi.Interface {
+	return c.gwApiCS
+}
+
+func (c *akoControlConfig) SetGatewayApiInformers(i *gwApiInformers) {
+	c.gwApiInformers = i
+}
+
+func (c *akoControlConfig) GatewayApiInformers() *gwApiInformers {
+	return c.gwApiInformers
+}
+
+func (c *akoControlConfig) ControllerVersion() string {
+	return c.controllerVersion
+}
+
+func (c *akoControlConfig) SetControllerVersion(v string) {
+	c.controllerVersion = v
+}
+
+func initControllerVersion() string {
+	version := os.Getenv("CTRL_VERSION")
+	if version == "" {
+		return version
+	}
+
+	// Ensure that the controllerVersion is less than the supported Avi maxVersion and more than minVersion.
+	if lib.CompareVersions(version, ">", lib.GetAviMaxSupportedVersion()) {
+		utils.AviLog.Infof("Setting the client version to AVI Max supported version %s", lib.GetAviMaxSupportedVersion())
+		version = lib.GetAviMaxSupportedVersion()
+		return version
+	}
+
+	if lib.CompareVersions(version, "<", lib.GetAviMinSupportedVersion()) {
+		AKOControlConfig().PodEventf(
+			corev1.EventTypeWarning,
+			lib.AKOShutdown, "AKO is running with unsupported Avi version %s",
+			version,
+		)
+		utils.AviLog.Fatalf("AKO is not supported for the Avi version %s, Avi must be %s or more", version, lib.GetAviMinSupportedVersion())
+	}
+	return ""
+}
+
+func (c *akoControlConfig) SetEventRecorder(id string, client kubernetes.Interface, fake bool) {
+	c.akoEventRecorder = utils.NewEventRecorder(id, client, fake)
+}
+
+func (c *akoControlConfig) EventsSetEnabled(enable string) {
+	if enable == "true" {
+		utils.AviLog.Infof("Enabling event broadcasting via AKO.")
+		c.akoEventRecorder.Enabled = true
+	} else {
+		utils.AviLog.Infof("Disabling event broadcasting via AKO.")
+		c.akoEventRecorder.Enabled = false
+	}
+}
+
+func (c *akoControlConfig) EventRecorder() *utils.EventRecorder {
+	return c.akoEventRecorder
+}
+
+func (c *akoControlConfig) SaveAKOPodObjectMeta(pod *v1.Pod) {
+	c.akoPodObjectMeta = &pod.ObjectMeta
+}
+
+func (c *akoControlConfig) PodEventf(eventType, reason, message string, formatArgs ...string) {
+	if c.akoPodObjectMeta != nil {
+		if len(formatArgs) > 0 {
+			c.EventRecorder().Eventf(&v1.Pod{ObjectMeta: *c.akoPodObjectMeta}, eventType, reason, message, formatArgs)
+		} else {
+			c.EventRecorder().Event(&v1.Pod{ObjectMeta: *c.akoPodObjectMeta}, eventType, reason, message)
+		}
+	}
+}
+
+func GetResponseFromURI(client *clients.AviClient, uri string) (models.SystemConfiguration, error) {
+	response := models.SystemConfiguration{}
+	err := lib.AviGet(client, uri, &response)
+
+	if err != nil {
+		utils.AviLog.Warnf("Unable to fetch system configuration, error %s", err.Error())
+	}
+
+	return response, err
+}
```

Comment:
shall we reuse the function in AKO here?

Reply:
updated to reuse the function